internal/app/gateway/command/run.go

```diff
@@ -197,6 +220,15 @@ func reloadGatewayFromConfigurationFile() {
 	if err != nil {
 		log.Println("[mashling] error stopping gateway:", err)
 	}
+	pingData := PingData{gateway.Version(), gateway.AppVersion(), gateway.Description()}
```

Comment:
Same as above. I would just move this into the Gateway itself so it can handle all related service rules. It will also eliminate code duplication.